### PR TITLE
feat: record sst file time range in FileMeta

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

First step towards support pruning SST files accroding to time range in query: extract the time ranges of SST files on flush to manifest.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#846

### CODE REVIEW

File: src/storage/src/sst/parquet.rs [hunk 1]
```diff
@@ -109,14 +115,91 @@ impl<'a> ParquetWriter<'a> {
...
+            .columns
+            .get(ts_index)
+            .context(DecodeParquetTimeRangeSnafu {
+                msg: format!("Cannot find ts column by index: {ts_index}"),
+            })?
+            .meta_data else { return Ok(None) };
+        let Some(stats) = &metadata.statistics else { return Ok(None) };
+        let (Some(min_value), Some(max_value)) = (&stats.min_value, &stats.max_value) else { return Ok(None); };
+
+        let min = i64::from_le_bytes(min_value[..8].try_into().map_err(|_| {
```
Comment (killme2008) [inline src/storage/src/sst/parquet.rs:181]: is it always in little endian?

Comment (killme2008) [reply src/storage/src/sst/parquet.rs:181]: I think we should log the error, or put it into `DecodeParquetTimeRange`'s msg.

Comment (v0y4g3r) [reply src/storage/src/sst/parquet.rs:181]: > is it always in little endian?

As per [parquet's implementation](https://docs.rs/parquet/29.0.0/src/parquet/file/statistics.rs.html#171-177), it should always be little endian.

**Review by waynexia — APPROVED** [review]

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**